cmd/restic/cmd_prune.go

```diff
@@ -297,19 +299,51 @@ func pruneRepository(gopts GlobalOptions, repo restic.Repository) error {
 	}
 
 	if len(removePacks) != 0 {
-		bar = newProgressMax(!gopts.Quiet, uint64(len(removePacks)), "packs deleted")
-		bar.Start()
-		for packID := range removePacks {
-			h := restic.Handle{Type: restic.DataFile, Name: packID.String()}
-			err = repo.Backend().Remove(ctx, h)
-			if err != nil {
-				Warnf("unable to remove file %v from the repository\n", packID.Str())
-			}
-			bar.Report(restic.Stat{Blobs: 1})
-		}
-		bar.Done()
+		DeleteFiles(gopts, true, repo, removePacks, restic.DataFile)
 	}
 
 	Verbosef("done\n")
 	return nil
 }
+
+const numDeleteWorkers = 8
+
+// DeleteFiles deletes the given fileList of fileType in parallel
+// if ignoreError=true, it will print a warning if there was an error, else it will abort.
+func DeleteFiles(gopts GlobalOptions, ignoreError bool, repo restic.Repository, fileList restic.IDSet, fileType restic.FileType) error {
```

Comment:
As `DeleteFiles` is not only used for prune this function deserves its own file. Maybe something like `delete_files.go` or `parallel.go`?

Reply:
Ok, agree. I'll change this.

IMO we have a similar thing with `newProgressMax` in `cmd_prune.go`. Maybe also extract this (in another PR)? Where do you think could this fit?

Reply:
The most obvious place to extract `newProgressMax` to would be `cmd/restic/progress.go`. That should be done in a separate PR, especially as `newReadProgress` in `cmd_check` could then also be deduplicated.

Reply:
I used `delete.go`...